src/classes/CW_AudienceCSVUploadControllerTest.cls

```diff
@@ -0,0 +1,397 @@
+		audGr2.Name = group2;
+		audGr2.Is_Control_Group__c = false;
+		insert audGr2;
+		
+		CW_Audience_Group__c audGr5 = TestDataUtil.createAudienceGroup(camp.Id);
+		audGr5.Name = group5;
+		audGr5.Is_Control_Group__c = true;
+		insert audGr5;
+		
+		system.assertEquals(2,[select count() from CW_Audience_Group__c where Campaign__c = :camp.Id]);
+		
+		Test.setCurrentPage(Page.CW_AudienceCSVUpload);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_CAMPAIGNID,camp.Id);
+		
+		Test.startTest();
+		CW_AudienceCSVUploadController uc = new CW_AudienceCSVUploadController();
+		uc.selectedSchId = sch.Id;
+		uc.csvFile = Blob.valueOf(groupCSVString.replace(',',':'));	// replace commas with colons
+		uc.filename='Somefile.csv';
+		uc.uploadCSV();
+		Test.stopTest();
+		
+		// 3 new records were added because of the new separator
+		system.assertEquals(5,[select count() from CW_Audience_Group__c where Campaign__c = :camp.Id]);
+		// group 2 wasn't added again since was previously created
+		system.assertEquals(1,[select count() from CW_Audience_Group__c 
+								where Campaign__c = :camp.Id and Name = :group2]);
+		// group 5 wasn't added again since was previously created
+		system.assertEquals(1,[select count() from CW_Audience_Group__c 
+								where Campaign__c = :camp.Id and Name = :group5]);
+	}
+	
+	static testMethod void TestErrors_FileMising()
+	{
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+		insert camp;
+		CW_Schedule__c sch = TestDataUtil.createSchedule(camp);
+		insert sch;
+		
+		Test.setCurrentPage(Page.CW_AudienceCSVUpload);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_CAMPAIGNID,camp.Id);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_OBJECTTYPE,ConstantUtil.PAGEPARAMVAL_AUDIENCE_GROUP);
+		
+		CW_AudienceCSVUploadController uc = new CW_AudienceCSVUploadController();
+		uc.selectedSchId = sch.Id;
+		system.assertEquals(null,uc.uploadCSV());
+	  	system.assertEquals(ApexPages.getMessages()[0].getSummary(),System.Label.CW_Error_FileNameMissing);
+	}
+	
+ 	static testMethod void TestErrors_FileBinary()
+	{
+		CW_Campaign__c camp = TestDataUtil.createCampaign();
+		insert camp;
+		CW_Schedule__c sch = TestDataUtil.createSchedule(camp);
+		insert sch;
+		
+		Test.setCurrentPage(Page.CW_AudienceCSVUpload);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_CAMPAIGNID,camp.Id);
+		Apexpages.currentPage().getParameters().put(ConstantUtil.PAGEPARAM_OBJECTTYPE,ConstantUtil.PAGEPARAMVAL_AUDIENCE_GROUP);
+		
+		CW_AudienceCSVUploadController uc = new CW_AudienceCSVUploadController();
+		uc.selectedSchId = sch.Id;
+		uc.csvFile =Blob.valueOf(String.fromCharArray(new List<Integer>{ 80,75,3,34,1,1,1,1,1,11,1}));
+		uc.filename='Somefile.csv';
+		
+		system.assertEquals(null,uc.uploadCSV());
+		system.assertEquals(System.Label.CW_Error_BinaryFile,ApexPages.getMessages()[0].getSummary());
+
+	}
+	
+	static testMethod void TestErrors_InvalidChar()
```

Comment:
* I suggest to annotate test methods with `@IsTest` for better readability.
* Methods should use the `lowerCamelCase` naming convention. I suggest to rename accordingly.